flowtools/flowclr_sample_name/sample_simp.R

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/Rscript
```

Comment:
As a general remark, I'm not sure if this is the best way of achieving what you are after. In general in Galaxy, AFAIK, you cannot modify the order of the elements in a collection. I'm also not sure if the order of iteration respects the order in which they are displayed in the history.

Reply:
If this sample number depends on metadata of the upload (usually encoded in the file name), I would suggest instead to tag this data automatically with regexps on upload as explained [in this tutorial](https://galaxyproject.github.io/training-material/topics/galaxy-interface/tutorials/group-tags/tutorial.html). Then, you can use the tag as an input to the tool (also shown in that tutorial).

Reply:
Otherwise, please explain to me the use case to try to find the correct way of doing this.

Reply:
Now, if you only want to tell the datasets apart for later merging, but you don't care which dataset gets which number (although I would doubt it, since you would lose track of things), then something like this could be applied (although I would still be a bit at odds with it).